feature-libs/checkout/base/core/facade/checkout-payment.service.spec.ts

```
@@ -0,0 +1,338 @@
+        loading: true,
+        error: false,
+        data: undefined,
+      });
+
+      // 2*2*300 = 1200
+      tick(1200);
+      expect(resultState).toEqual({
+        loading: false,
+        error: false,
+        data: mockCardTypes,
+      });
+      subscription.unsubscribe();
+    }));
+  });
+
+  describe(`getCardTypes`, () => {
+    it(`should call facade's getCardTypesState()`, (done) => {
+      spyOn(service, 'getCardTypesState').and.returnValue(
+        of({
+          loading: false,
+          error: false,
+          data: mockCardTypes,
+        })
+      );
+
+      service
+        .getCardTypes()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(mockCardTypes);
+          done();
+        });
+    });
+
+    it(`should return an empty array if query's data is falsy`, (done) => {
+      spyOn(service, 'getCardTypesState').and.returnValue(
+        of({
+          loading: false,
+          error: false,
+          data: undefined,
+        })
+      );
+
+      service
+        .getCardTypes()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual([]);
+          done();
+        });
+    });
+  });
+
+  describe(`getPaymentDetailsState`, () => {
+    it(`should return the delivery modes`, (done) => {
+      spyOn(checkoutQuery, 'getCheckoutDetailsState').and.returnValue(
+        of(<QueryState<CheckoutState>>{
+          loading: false,
+          error: false,
+          data: {
+            paymentInfo: mockPaymentInfo,
+          },
+        })
+      );
+
+      service
+        .getPaymentDetailsState()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(<QueryState<PaymentDetails | undefined>>{
+            loading: false,
+            error: false,
+            data: mockPaymentInfo,
+          });
+          done();
+        });
+    });
+  });
+
+  describe(`createPaymentDetails`, () => {
```

add 2 `it`

1 - should throw error when paymentDetails is undefined and expect no connector calls
2 - should not dispatch LoadUserPaymentMethods(xxx) when anonymous user

---

1 - removed the checks, as it shouldn't be possible to pass a falsy value
2 - added